lighthouse-core/audits/seo/json-ld.js

```diff
@@ -0,0 +1,172 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit.js');
+const validateJsonLD = require('../../lib/sd-validation/sd-validation.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether JSON-LD structured data snippets are valid. This descriptive title is shown when no invalid JSON-LD snippets were found. */
+  title: 'JSON-LD structured data syntax is valid',
+  /** Title of a Lighthouse audit that provides detail on whether JSON-LD structured data snippets are valid. This descriptive title is shown when JSON-LD snippets with invalid content were found. */
+  failureTitle: 'JSON-LD structured data syntax is invalid',
+  /** Description of a Lighthouse audit that tells the user whether JSON-LD snippets on the page are invalid. This is displayed after a user expands the section to see more. No character length limits. */
+  /* eslint-disable-next-line max-len */
+  description: 'Structured data contains rich metadata about a web page. The data is used in search results and social sharing. Invalid metadata will affect how the page appears in these contexts. This audit currently validates a subset of JSON-LD rules. See also the [Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool/) to validate other types of structured data.',
+  /** Explanatory message stating how many JSON-LD structured data snippets are invalid */
+  displayValue: `{invalidSnippetCount, plural,
+    =1 {# invalid snippet}
+    other {# invalid snippets}
+    }`,
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class StructuredDataAutomatic extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'json-ld',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['ScriptElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
+    const jsonLDElements = artifacts.ScriptElements.filter(
+      script => script.type === 'application/ld+json' && !!script.content);
+
+    if (jsonLDElements.length === 0) {
+      return {
+        notApplicable: true,
+        score: 1,
+      };
+    }
+
+    const validatedSnippets = await Promise.all(jsonLDElements.map(async (element) => {
```

Comment:
`validatedSnippets` is kind of confusing here as a variable name (e.g. invalid validatedSnippets breaks my brain a little)

Just `snippets`, maybe? Doesn't seem to lose out on context below (a snippet in `snippets` can either be valid or invalid)